sksurv/tree/_criterion.pyx

```diff
@@ -328,19 +330,31 @@ cdef class LogrankCriterion(Criterion):
             DOUBLE_t ratio
             DOUBLE_t n_events
             DOUBLE_t n_at_risk
-
-        self.riskset_total.at(0, &n_at_risk, &n_events)
-        ratio = n_events / n_at_risk
-        dest[0] = ratio  # Nelson-Aalen estimator
-        dest[1] = 1.0 - ratio  # Kaplan-Meier estimator
-
-        j = 2
-        for i in range(1, self.n_unique_times):
-            self.riskset_total.at(i, &n_at_risk, &n_events)
-            dest[j] = dest[j - 2]
-            dest[j + 1] = dest[j - 1]
-            if n_at_risk != 0:
-                ratio = n_events / n_at_risk
-                dest[j] += ratio
-                dest[j + 1] *= 1.0 - ratio
-            j += 2
+            DOUBLE_t dest_j0
+
+        # low memory mode
+        if  self.n_outputs == 1:
+            dest[0] = dest_j0 = 0
+            for i in range(0, self.n_unique_times):
+                self.riskset_total.at(i, &n_at_risk, &n_events)
+                if n_at_risk != 0:
+                    ratio = n_events / n_at_risk
+                    dest_j0 += ratio
+                if self.is_event_time[i]:
```

Comment:
You should assign `self.is_event_time` a local variable first, because otherwise each call will have some overhead. You can see that if you run `cython -a _criterion.pyx` and have a look the generated HTML file. If a line is highlighted yellow, there is some overhead.